### DISCUSSION TITLE
How to pop up a window in mobile?

### DISCUSSION THREAD

**Discussion by j4587698** [AvaloniaUI/Avalonia#9984] [Q&A]
How to pop up a window like android popwindow?

**Comment by timunie** [comment]
I think this is not yet implemented but planned, see: https://github.com/AvaloniaUI/Avalonia/discussions/7929#discussioncomment-4692189

You can still write native code and use a service to call it. I have no example at hand though.

**Reply by j4587698** [reply]
Thank you for your reply. This discussion is about a MessageBox, but I want a Dialog that can display complex controls.

**Reply by timunie** [reply]
In that case check out what FluentAvalonia does. I'm not sure if it works for mobile, but may be a good starting point. 

https://github.com/amwx/FluentAvalonia/tree/master/FluentAvalonia/UI/Controls/ContentDialog 
https://github.com/amwx/FluentAvalonia/tree/master/FluentAvalonia/UI/Controls/TaskDialog

**Reply by maxkatz6** [reply]
@timunie I don't think it's related. As MessageBox is a very specific control, while this question, as it seems, about more generic Window support on the mobile. Which won't come in the future, as there are no windows there (we possibly will have Scene alternative from the iOS though, but that's not for 11.0).

@j4587698 basically, you need to use Popup. FluentAvalonia.ContentDialog is a good example, as it uses popups internally. But for the simple use cases Popup API isn't difficult.

**Reply by maxkatz6** [reply]
Another option, as @lukaszossowski mentioned, is to create an overlay layer. Just add two panels on top of each other, for example. It will work for many cases just fine too.
Currently, Popup actually works in the build-in overlay layer on the mobile, so there won't be much difference under the hood.

**Comment by lukaszossowski** [comment]
The way I do it currently is by having an overlay control that sits on top of the app, invisible and hittest invisible as well, and when I need to display something complex I activate it.
